packages/components/tests/dummy/app/templates/components/modal.hbs

```diff
@@ -0,0 +1,669 @@
+  </p>
+
+  <h4 class="dummy-h4">Form within a modal dialog</h4>
+  {{! prettier-ignore-start }}
+  {{! template-lint-disable no-unbalanced-curlies }}
+  <CodeBlock
+    @language="markup"
+    @code='
+      <Hds::Button @text="Open form modal" @color="secondary" \{{on "click" (fn this.activateModal "formModalActive")}} />
+
+      \{{#if this.formModalActive}}
+        <Hds::Modal id="form-modal" @onClose=\{{fn this.deactivateModal "formModalActive"}} as |M|>
+          <M.Header>
+            Why do you want to leave the beta?
+          </M.Header>
+          <M.Body>
+            <Hds::Form::Select::Field autofocus @width="100%" as |F|>
+              <F.Label>Select the primary reason</F.Label>
+              <F.Options>
+                <option></option>
+              </F.Options>
+            </Hds::Form::Select::Field>
+            <Hds::Form::Textarea::Field @isOptional=\{{true}} as |F|>
+              <F.Label>Your feedback</F.Label>
+            </Hds::Form::Textarea::Field>
+          </M.Body>
+          <M.Footer>
+            <Hds::ButtonSet>
+              <Hds::Button type="submit" @text="Leave Beta" />
+              <Hds::Button type="submit" @text="Cancel" @color="secondary" \{{on "click" (fn this.deactivateModal "formModalActive")}} />
+            </Hds::ButtonSet>
+          </M.Footer>
+        </Hds::Modal>
+      \{{/if}}
+    '
+  />
+  {{! template-lint-enable no-unbalanced-curlies }}
+  {{! prettier-ignore-end }}
+  <p class="dummy-paragraph">Renders to:</p>
+  <Hds::Button @text="Open form modal" @color="secondary" {{on "click" (fn this.activateModal "formModalActive")}} />
+
+  {{! template-lint-disable no-autofocus-attribute }}
+  {{#if this.formModalActive}}
+    <Hds::Modal id="form-modal" @onClose={{fn this.deactivateModal "formModalActive"}} as |M|>
+      <M.Header>
+        Why do you want to leave the beta?
+      </M.Header>
+      <M.Body>
+        <Hds::Form::Select::Field autofocus @width="100%" as |F|>
+          <F.Label>Select the primary reason</F.Label>
+          <F.Options>
+            <option></option>
+          </F.Options>
+        </Hds::Form::Select::Field>
+        <Hds::Form::Textarea::Field @isOptional={{true}} as |F|>
+          <F.Label>Your feedback</F.Label>
+        </Hds::Form::Textarea::Field>
+      </M.Body>
+      <M.Footer>
+        <Hds::ButtonSet>
+          <Hds::Button type="submit" @text="Leave Beta" />
+          <Hds::Button
+            type="submit"
+            @text="Cancel"
+            @color="secondary"
+            {{on "click" (fn this.deactivateModal "formModalActive")}}
+          />
+        </Hds::ButtonSet>
+      </M.Footer>
+    </Hds::Modal>
+  {{/if}}
+  {{! template-lint-enable no-autofocus-attribute }}
+  <p class="dummy-banner dummy-banner--info dummy-paragraph"><FlightIcon @name="info" />
+    Note: If a modal dialog contains interactive elements, such as a form, the initial focus should be set on the first
```

Comment:
[question] where is this guidance coming from?

[suggestion] I would probably try to re-word this to be more explicit about the context.

Reply:
From memory, based on [the W3C example used as a reference](https://www.w3.org/WAI/ARIA/apg/example-index/dialog-modal/dialog.html) 😬 
Happy to try a re-word, but if you have some time please adjust as needed and I'll commit it.